main.cpp

```diff
@@ -0,0 +1,238 @@
+#include <iostream>
+
+#include "student.h"
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+using std::back_inserter;
+using std::cin;
+using std::count_if;
+using std::cout;
+using std::endl;
+using std::find;
+using std::istream;
+using std::less;
+using std::map;
+using std::ostream;
+using std::pair;
+using std::replace_if;
+using std::sort;
+using std::string;
+using std::to_string;
+using std::transform;
+using std::unordered_multimap;
+using std::unordered_set;
+using std::vector;
+
+template<typename E>
+string to_string(const vector<E> vector) {
+    string value = "{";
+    {
+        auto not_first = false;
+        for (auto &element : vector) {
+            if (not_first) value += ", ";
+            else not_first = true;
+
+            value += element;
+        }
+    }
+    return value + "}";
+}
+
+template<typename E>
+string to_string(const unordered_set<E> vector) {
+    string value = "{";
+    {
+        auto not_first = false;
+        for (auto &element : vector) {
+            if (not_first) value += ", ";
+            else not_first = true;
+
+            value += element;
+        }
+    }
+    return value + "}";
+}
+
+template<typename K, typename V>
+string to_string(const map<K, V> map) {
+    string value = "{";
+    {
+        const auto end = map.end();
+        auto not_first = false;
+        for (auto iterator = map.begin(); iterator != end; iterator++) {
+            if (not_first) value += ", ";
+            else not_first = true;
+
+            const auto pair = *iterator;
+            value += (to_string(pair.first) + ":" + pair.second);
+        }
+    }
+    return value + "}";
+}
+
+template<typename K, typename V>
+string to_string(const unordered_multimap<K, V> map) {
+    string value = "{";
+    {
+        const auto end = map.end();
+        auto not_first = false;
+        for (auto iterator = map.begin(); iterator != end; iterator++) {
+            if (not_first) value += ", ";
+            else not_first = true;
+
+            const auto pair = *iterator;
+            value += (to_string(pair.first) + ":" + pair.second);
+        }
+    }
+    return value + "}";
+}
+
+void task_lambda();
+
+void task_io();
+
+int main() {
+    unordered_multimap<int, string> original_map;
+
+    while (true) {
+        cout << "Select task group (`lambda`, `io` or `exit`):" << endl;
+        string mode;
+        cin >> mode;
+
+        if (mode == "lambda") task_lambda();
+        else if (mode == "io") task_io();
+        if (mode == "exit") {
+            cout << "Goodbye!" << endl;
+            return 0;
+        } else cout << "Unknown mode" << endl;
+    }
+}
+
+void task_lambda() {
+    unordered_multimap<int, string> original_map;
```

Comment:
The unordered multimap was already created in main(), why u do it?

Reply:
Resolved in 681a7cf